hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DeploymentManagementTest.java

```diff
@@ -1053,18 +1074,22 @@ public void handleEvent(final TargetAssignDistributionSetEvent event) {
         }
     }
 
-    private static class CancelEventHandlerMock {
+    private static class CancelEventHandlerStub {
         private final List<CancelTargetAssignmentEvent> events = Collections.synchronizedList(new LinkedList<>());
-        private final CountDownLatch latch;
-        private final int expectedNumberOfEvents;
+        private CountDownLatch latch;
+        private int expectedNumberOfEvents;
 
-        private CancelEventHandlerMock(final int expectedNumberOfEvents) {
+        public void setExpectedNumberOfEvents(final int expectedNumberOfEvents) {
+            events.clear();
```

Comment:
Why CancelEventHandlerStub gets registered as bean and the setter is called multiple times in different test cases. By setter you can change the values of the bean at runtime, with constructor only by initialization.